tests/test_qiskit_device.py

```diff
@@ -45,72 +22,22 @@ def test_warning_raised_for_hardware_backend_analytic_expval(self, hardware_back
 
         # check that only one warning was raised
         assert len(record) == 1
         # check that the message matches
-        assert record[0].message.args[0] == "The analytic calculation of expectations and variances "\
-                                            "is only supported on statevector backends, not on the {}. "\
-                                            "The obtained result is based on sampling.".format(dev.backend)
+        assert record[0].message.args[0] == "The analytic calculation of "\
+                "expectations, variances and probabilities is only supported on "\
+                "statevector backends, not on the {}. Such statistics obtained from this "\
+                "device are estimates based on samples.".format(dev.backend)
 
     def test_no_warning_raised_for_software_backend_analytic_expval(self, statevector_backend, recorder, recwarn):
         """Tests that no warning is raised if the analytic attribute is true on
             statevector simulators when calculating the expectation"""
 
         dev = qml.device("qiskit.basicaer", backend=statevector_backend, wires=2, analytic=True)
 
-        @qml.qnode(dev)
-        def circuit():
-            qml.Hadamard(wires=0)
-            return qml.expval(qml.PauliZ(0))
-
-        circuit()
-
-        # check that no warnings were raised
-        assert len(recwarn) == 0
-
-    def test_warning_raised_for_hardware_backend_analytic_var(self, hardware_backend, recorder):
-        """Tests that a warning is raised if the analytic attribute is true on
-            hardware simulators when calculating the variance"""
-
-        dev = qml.device("qiskit.basicaer", backend=hardware_backend, wires=2, analytic=True)
-
-        @qml.qnode(dev)
-        def circuit():
-            qml.Hadamard(wires=0)
-            return qml.var(qml.PauliZ(0))
-
-        with pytest.warns(UserWarning) as record:
-            circuit()
-
-        # check that only one warning was raised
-        assert len(record) == 1
-        # check that the message matches
-        assert record[0].message.args[0] == "The analytic calculation of expectations and variances "\
-                                            "is only supported on statevector backends, not on the {}. "\
-                                            "The obtained result is based on sampling.".format(dev.backend)
-
-    def test_no_warning_raised_for_software_backend_analytic_var(self, statevector_backend, recorder, recwarn):
-        """Tests that no warning is raised if the analytic attribute is true on
-            statevector simulators when calculating the variance"""
-
-        dev = qml.device("qiskit.basicaer", backend=statevector_backend, wires=2, analytic=True)
-
-        @qml.qnode(dev)
-        def circuit():
-            qml.Hadamard(wires=0)
-            return qml.var(qml.PauliZ(0))
-
-        circuit()
-
         # check that no warnings were raised
         assert len(recwarn) == 0
```

Comment:
What exactly is ``recwarn``? It's not in the import, nor in the ``conftest.py`` file? Seems to be very useful if it just collects warnings!

Reply:
Yes! :) It is one of the [default fixtures from `pytest`](https://docs.pytest.org/en/latest/warnings.html#recwarn). It is similar to `monkeypatch` that we frequently use. Other fixture useful for capturing output with `pytest` are `capsys`, `capfd` for example.

Reply:
Nice, thanks, will look them up!